pubsub/azure/servicebus/servicebus.go

```diff
@@ -360,6 +369,62 @@ func (a *azureServiceBus) Publish(req *pubsub.PublishRequest) error {
 }
 
 func (a *azureServiceBus) Subscribe(subscribeCtx context.Context, req pubsub.SubscribeRequest, handler pubsub.Handler) error {
+	sub := impl.NewSubscription(
+		subscribeCtx,
+		a.metadata.MaxActiveMessages,
+		a.metadata.TimeoutInSec,
+		a.metadata.MaxRetriableErrorsPerSec,
+		a.metadata.MaxConcurrentHandlers,
+		"topic "+req.Topic,
+		a.logger,
+	)
+
+	receiveAndBlockFn := func(bo *backoff.ExponentialBackOff) error {
```

Comment:
Nit: rather than passing `bo` as an argument and then creating `func() { bo.Reset()}`, pass directly a function (defined by the caller) that performs `bo.Reset()`